common/utils/backend.py

```diff
@@ -357,6 +359,7 @@ def get_next_num_run(self, peek: bool = False) -> int:
         other_num_runs = [
             int(os.path.basename(run_dir).split("_")[1])
             for run_dir in glob.glob(os.path.join(self.internals_directory, "runs", "*"))
+            if self._is_run_dir(os.path.basename(run_dir))
         ]
```

Comment:
Not an issue, just a statement:

The whole function, if this is the only place it's used could be just
```python
        other_num_runs = [
            int(os.path.basename(run_dir).split("_")[1])
            for run_dir in glob.glob(os.path.join(self.internals_directory, "runs", "*"))
            if re.match(r"\d+_\d+_\d+", run_dir)
        ]
```

Saves from adding extra functions but maybe it's good to have the functionality encapsulated somewhere.

Reply:
yeah I had it this way previously, but I think this makes it a bit more readable to the user.